### PR TITLE
Begin payload reduction

### PR DESCRIPTION
**Pull request by schwers**
This is the first in a series of payload optimizations. 

For an overview (@madbook, this should help explain @curioussavage's patch) please see: https://github.com/schwers/reddit-mobile/pull/3 
* note: these numbers are somewhat outdated as there has since been additional commits to mweb, and we're now including raven. 

👓  @curioussavage (cc @madbook, @birakattack)

### Commits
The first commit in this patch sets us up to do analysis step by step, and have a visualization to guide our efforts.

The second commit is "the low hanging fruit" of potential optimizations we can make.

### Results
Before and After:
* 357 KB gzipped -> 264 KB gzipped
* 1.3 MB -> 1017 KB

This work, and the rest of the work in the bundle audit can be visualized with `webpack-bundle-analyzer`. With the first commit you can run `npm run analyze` to generate interactive visualizations locally. The sizing information it prints, is slightly different than the numbers you'll see in chrome. The number's I'm referencing in the these results are from gzipping `ProductionClient` and running `npm run start`, and using the stats from Chrome's `Network` tab.

Before this patch, you can see a large chunks of: `elliptic`, `bn`(big nums), `browserify-aes`, `asn1`, `parse-asn1`, `hash`, `sha`, `des`, `diffie-hellmen`, node's `Buffer`

image

After this patch's second commit, that bulk is removed:

image

### CODE REVIEW

**Comment by madbook** [comment]
🔕 This is cool stuff! Is there a ticket linked to this?

**Comment by curioussavage** [comment]
Technically its lumped in with this ticket just like my redux archiver one. https://reddit.atlassian.net/browse/CONT-163

There will be at least one more pull request we extract from @schwers work. Just nicer to test/review it in parts.

File: src/server/meta/index.js [hunk 1]
```diff
@@ -196,16 +196,8 @@ export default (router, apiOptions) => {
...
-
-    const hmac = crypto.createHmac(algorithm, secret);
-    hmac.setEncoding('hex');
-    hmac.write(timings.actionName);
-    hmac.end();
-
-    const hash = hmac.read();
-
+    const secret = atob(apiOptions.actionNameSecret);
+    const hash =  HmacSha1(timings.actionName, secret);
```
Comment (curioussavage) [inline src/server/meta/index.js]: looks like the space after equals is the cause of the linter error

**Comment by curioussavage** [comment]
Only concern is if the output is still the same for the modified functions.

I was able to login while testing this.
💅

**Comment by curioussavage** [comment]
woops, didn't mean to close this.

**Comment by schwers** [comment]
I made a branch you can use to verify these changes produce the same values as before. Testing relies on running that code locally, and verifying in the debug console. I added an explicit call of the amp link generation to test, as it seems `createAmpHtmlLinkFromState` has a minor bug (I'll make a separate ticket).

https://github.com/schwers/reddit-mobile/commit/b7dc4dbc7baec5ac5577e942e12f4ff8583b199e

:haircut_man:

**Comment by curioussavage** [comment]
I just ran your test branch. everything looks good so far.

**Comment by curioussavage** [comment]
🐟

**Pull request merged**